scripts/jest_integration.js

```diff
@@ -40,6 +40,11 @@
 //
 // See all cli options in https://jestjs.io/docs/cli
 
+/* Increase the max listeners
+ * Numerous listeners are added by `root.start()` that are never removed.
+ */
+process.setMaxListeners(Infinity);
```

Comment:
We have fixed the memory leak here https://github.com/opensearch-project/OpenSearch-Dashboards/issues/3601#issuecomment-1478405674

Increasing the limit for listeners won't itself directly cause memory leaks; it only allows more listeners to be added without warnings. The potential memory leak comes from the accumulation of listeners that aren't properly cleaned up. If removing them when they're no longer needed properly, then increasing the limit should not exacerbate memory issues.

We have a large test suite with many async operations, it's possible that you're simply exceeding the default limit of 10 listeners without actually leaking memory. So we want to catch and fix leaking memory and also increase the listeners.

Reply:
We could set it to 20 `process.setMaxListeners(20);`
If don't set it will get the following in the integ tests
```
(node:3286593) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 unhandledRejection listeners added to [process]. Use emitter.setMaxListeners() to increase limit
(Use `node --trace-warnings ...` to show where the warning was created)
(node:3286593) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 warning listeners added to [process]. Use emitter.setMaxListeners() to increase limit
```

Reply:
Keep in mind that this is only within tests and tests add numerous listeners or call functions that add listeners numerous times.